### PR TITLE
tran_sock: don't leave errno unset when failing to attach

### PR DESCRIPTION
**Pull request by tmakatos**
Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -437,6 +437,8 @@ handle_device_get_info(vfu_ctx_t *vfu_ctx, uint32_t in_size,
 
     if (in_size < sizeof(*in_dev_info) ||
         in_dev_info->argsz < sizeof(*in_dev_info)) {
+        vfu_log(vfu_ctx, LOG_ERR, "bad device get info size %d %d", in_size,
```
Comment (jlevon) [inline lib/libvfio-user.c:440]: unrelated change?

**Review by jlevon — CHANGES_REQUESTED** [review]
Thanks, d'oh.

File: lib/libvfio-user.c (hunk 1 above)
Comment (tmakatos) [reply lib/libvfio-user.c:440]: yep, sorry

**Review by jlevon — APPROVED** [review]
